x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/PreAnalyzer.java

```diff
@@ -47,17 +55,19 @@ public PreAnalysis preAnalyze(LogicalPlan plan) {
     protected PreAnalysis doPreAnalyze(LogicalPlan plan) {
         List<TableInfo> indices = new ArrayList<>();
         List<Enrich> unresolvedEnriches = new ArrayList<>();
+        List<InferencePlan> unresolvedInferencePlans = new ArrayList<>();
         List<TableInfo> lookupIndices = new ArrayList<>();
 
         plan.forEachUp(UnresolvedRelation.class, p -> {
             List<TableInfo> list = p.indexMode() == IndexMode.LOOKUP ? lookupIndices : indices;
             list.add(new TableInfo(p.indexPattern()));
         });
         plan.forEachUp(Enrich.class, unresolvedEnriches::add);
+        plan.forEachUp(InferencePlan.class, unresolvedInferencePlans::add);
```

Comment:
ℹ️ Colllecting all InferencePlan during pre-analysis, so we know early what inference enpoints are required to execute the pipeline.